src/System.CommandLine/Parsing/ArgumentResult.cs

```diff
@@ -105,11 +105,12 @@ private ArgumentConversionResult Convert(Argument argument)
             if (ShouldCheckArity() &&
                 Parent is { } &&
                 ArgumentArity.Validate(Parent,
-                                       argument,
-                                       argument.Arity.MinimumNumberOfValues,
-                                       argument.Arity.MaximumNumberOfValues) is FailedArgumentConversionResult failedResult)
+                    argument,
+                    argument.Arity.MinimumNumberOfValues,
+                    argument.Arity.MaximumNumberOfValues) is ArgumentConversionResult failed &&
+                    failed is not null) // returns null on success
```

Comment:
`failed is not null` check isn't needed. I think it's redundant with `is ArgumentConversionResult failed` which won't match if the return value of `Validate` is `null`.